src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,194 @@
+                {
+                    var statementSyntax = blockStatements[0];
+                    return statementSyntax switch
+                    {
+                        // Note: For this case this will return null in Callee()
+                        // void Caller() { Callee(); }
+                        // void Callee() { return; }
+                        // Refactoring won't be provided for this case.
+                        ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                        ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                        _ => null
+                    };
+                }
+            }
+            else
+            {
+                // 2. If it is an Arrow Expression
+                var arrowExpressionNode = methodDeclarationSyntax.ExpressionBody;
+                if (arrowExpressionNode != null)
+                {
+                    return arrowExpressionNode.Expression;
+                }
+            }
+
+            return null;
+        }
+
+        protected override SyntaxNode? GetEnclosingMethodLikeNode(SyntaxNode syntaxNode)
+        {
+            for (var node = syntaxNode; node != null; node = node.Parent)
+            {
+                if (node.IsKind(SyntaxKind.MethodDeclaration)
+                    || node.IsKind(SyntaxKind.LocalFunctionStatement)
+                    || node is LambdaExpressionSyntax)
+                {
+                    return node;
+                }
+            }
+
+            return null;
+        }
+
+        protected override SyntaxNode GenerateTypeSyntax(ITypeSymbol symbol, bool allowVar)
+            => symbol.GenerateTypeSyntax(allowVar);
+
+        protected override ExpressionSyntax Parenthesize(ExpressionSyntax expressionSyntax)
+            => expressionSyntax.Parenthesize();
+        protected override bool IsValidExpressionUnderStatementExpression(ExpressionSyntax expressionNode)
+        {
+            // C# Expression Statements defined in the language reference
+            // expression_statement
+            //     : statement_expression ';'
+            //     ;
+            //
+            // statement_expression
+            //     : invocation_expression
+            //     | null_conditional_invocation_expression
+            //     | object_creation_expression
+            //     | assignment
+            //     | post_increment_expression
+            //     | post_decrement_expression
+            //     | pre_increment_expression
+            //     | pre_decrement_expression
+            //     | await_expression
+            //     ;
+            var isNullConditionalInvocationExpression =
+                expressionNode is ConditionalAccessExpressionSyntax conditionalAccessExpressionSyntax
+                && conditionalAccessExpressionSyntax.WhenNotNull.IsKind(SyntaxKind.InvocationExpression);
```

Comment:
that won't work for `a?.b?.C()`

Reply:
Changed this part to recursively checking ConditionalAccessExpressionSyntax